maestrowf/utils.py

```diff
@@ -168,6 +168,13 @@ def start_process(cmd, cwd=None, env=None, shell=True):
     if isinstance(cmd, list):
         shell = False
 
+    kwargs_to_Popen = {"shell": shell}
```

Comment:
My take on this:

```
def start_process(cmd, cwd=None, env=None, shell=True):
    """
    Starts a new process using a specified command.

    :param cmd: A string or a list representing the command to be run.
    :param cwd: Current working path that the process will be started in.
    :param env: A dictionary containing the environment the process will use.
    :param shell: Boolean that determines if the process will run a shell.
    """
    if isinstance(cmd, list):
        shell = False

    # Define kwargs for the upcoming Popen call.
    kwargs = {
        "shell":                shell,
        "cwd":                  cwd,
        "env":                  env,
        "universal_newlines":   True,
        "stdout":               PIPE,
        "stderr":               PIPE,
    }

    return Popen(cmd, **kwargs)
```

Though your code has the benefit of not passing in extra parameters.